### PR TITLE
@W-8881273 - Clean up webfiles based on learnings in training

### PR DESCRIPTION
**Pull request by kritikapradhan**

### CODE REVIEW

File: pkg/sloop/webserver/webfiles.go [hunk 1]
```diff
@@ -6,33 +6,29 @@ import (
...
 	if err == nil {
 		return data, err
-	}
-	files := AssetNames()
-	//if file exists in binary form
-	if common.Contains(files, filepath) {
-		return Asset(filepath)
 	} else {
-		return nil, err
+		//file exists in binary
```
Comment (duke-harlan) [inline pkg/sloop/webserver/webfiles.go:25]: `else {` above is not needed and can reduce indentation of following code

File: pkg/sloop/webserver/webfiles.go [hunk 2]
```diff
@@ -6,33 +6,29 @@ import (
...
+		binFileList := AssetNames()
+		if common.Contains(binFileList, common.GetFilePath(prefix, fileName)) {
```
Comment (duke-harlan) [inline pkg/sloop/webserver/webfiles.go:27]: consider a local variable `filepath := common.GetFilePath(prefix, fileName)` so they are combined only once

File: pkg/sloop/webserver/webfiles.go [hunk 3]
```diff
@@ -41,14 +37,11 @@ func readWebfile(filepath string, fs *afero.Afero) ([]byte, error) {
...
-	data, err := readWebfile((path.Join(prefix, templateName)), &fs)
-	if err != nil {
-		return nil, err
-	}
-	newTemplate := template.New(templateName)
-	newTemplate, err = newTemplate.Parse(string(data))
-	if err != nil {
-		return nil, err
+	data, err := readWebfile(templateName, &fs)
+	if err == nil {
```
Comment (duke-harlan) [inline pkg/sloop/webserver/webfiles.go:41]: the usual pattern of `if err != nil {` should be followed unless strong reason not too. It leaves the main line code at the least indentation

File: pkg/sloop/webserver/webfiles.go [hunk 4]
```diff
@@ -41,14 +37,11 @@ func readWebfile(filepath string, fs *afero.Afero) ([]byte, error) {
...
+		newTemplate := template.New(templateName)
+		newTemplate, err = newTemplate.Parse(string(data))
```
Comment (duke-harlan) [inline pkg/sloop/webserver/webfiles.go:43]: this `err` is being ignored

File: pkg/sloop/webserver/webserver.go [hunk 5]
```diff
@@ -99,8 +100,8 @@ func webFileHandler(currentContext string) http.HandlerFunc {
 			logWebError(nil, "Not allowed", r, w)
 			return
 		}
-		fullPath := path.Join(prefix, fixedUrl)
-		data, err := readWebfile(fullPath, &afero.Afero{afero.NewOsFs()})
+		fullPath := common.GetFilePath(prefix, fixedUrl)
```
Comment (duke-harlan) [inline pkg/sloop/webserver/webserver.go:103]: if `fullPath` still needed?

**Review by duke-harlan — COMMENTED** [review]
handful of comments in line

File: pkg/sloop/webserver/webserver.go (hunk 5 above)
Comment (kritikapradhan) [reply pkg/sloop/webserver/webserver.go]: yes still needed, i moved it right above the line it's being used.

File: pkg/sloop/webserver/webfiles.go [hunk 6]
```diff
@@ -6,33 +6,29 @@ import (
...
+// sample input : fileName="index.html"
+func readWebfile(fileName string, fs *afero.Afero) ([]byte, error) {
+	//file exists as a physical file
+	data, err := fs.ReadFile(common.GetFilePath(webFilesPath, fileName))
+	if err != nil {
+		//file exists in binary
+		binFileList := AssetNames()
+		binFileName := common.GetFilePath(prefix, fileName)
+		if common.Contains(binFileList, binFileName) {
+			return Asset(binFileName)
```
Comment (duke-harlan) [inline pkg/sloop/webserver/webfiles.go:27]: could all of this block of code be condenced to:
```
    return Asset(common.GetFilePath(prefix, fileName))
```
this is after seeing the `Asset()` is checking filenames, and assuming that the error message works (not confusing & contains the asset name)

**Review by duke-harlan — APPROVED** [review]
looks ok - 1 suggestion
- do we need to check the existence of the binary form before calling `Asset()`

**Pull request merged**